### ISSUE TITLE
undefined octokit                

### ISSUE THREAD

**Issue by martinch-kth** [octokit/octokit.js#1154]
I'm not sure it's an issue, I suspect my use of octokit and 'github-app' makes some kind of mess.

This works: octokit.issues.createComment({
But not this: octokit.checks.create({     --> create is undefined 

This is just a messy example of what I've tried: [mylink](
https://gist.github.com/martinch-kth/791714087f2b75d1ea2a038c76ab60da)

btw, where is the best place to ask questions around octokit/rest? Thanks.

**Comment by gr2m** [comment]
The octokit you get here
https://gist.github.com/martinch-kth/791714087f2b75d1ea2a038c76ab60da#file-gistfile1-txt-L29

Is not the octokit you load from line 1.

What you need is https://github.com/octokit/octokit.js which integrates GitHub Apps Authentication, webhook handling and the REST API, but it’s not ready yet. Have you checked out Probot: https://probot.github.io/ – it does the same thing and more

**Comment by gr2m** [comment]
I’ll close the issue because it’s out of scope for this repository, but please comment below if you have more questions

**Comment by martinch-kth** [comment]
Thanks. I will trying out probot for now. I've managed to use: github.repos.createCommitComment

Question:
I know this is kind of impossible, but I would like to update the commit message in all cases: push/commit/pullrequest.. I've read that you can't update a commit message since it's part of the commits sha/id. 

Do you know any hack to do so, just using the GitHub API? (besides using git to do it, which I've seen is possible.)

Thanks for the reply, really encouraging since I've been really stuck! (REALLY)

**Comment by gr2m** [comment]
Yes you cannot reword commit messages unfortunately, you’ll have to delete and create them again.

Just so I understand, you want to build a GitHub app that listens to every new commit on the default branch (usually `master`) and then change the commit message? Can you ellaborate why? Maybe there is a better way?

**Comment by martinch-kth** [comment]
Thanks, I used the status api instead. Any good irc channel for probot/github apps? Forums are great also but the github forum seemed a little bit slow? :-) Anyways, I did this octokit t-shirt to have on when coding to get in the mood... maybe you like to see it, its done with 2 layers cut out vinyl. :-) 
![img_20181216_001028](https://user-images.githubusercontent.com/43812208/51031816-038b7d00-159e-11e9-8677-50c518caf7e7.jpg)
